Geometry_Engine/Query/ClosestPoint.cs

```diff
@@ -106,6 +136,37 @@ public static Point ClosestPoint(this Circle circle, Point point, double toleran
 
         /***************************************************/
 
+        [Description("Finds the closest point on the Ellipse in relation to the provided Point.")]
+        [Input("ellipse", "The Ellipse on which to find the closest point in relation to the provided point.")]
+        [Input("point", "The reference point. The point on the Ellipse with the smallest distance to this point will be returned.")]
+        [Input("tolerance", "Distance tolerance to be used in the method.", typeof(Length))]
+        [Output("closePt", "The point on the Ellipse with the smallest distance to the provided point.")]
+        public static Point ClosestPoint(this Ellipse ellipse, Point point, double tolerance = Tolerance.Distance)
+        {
+            if (ellipse.IsNull() || point.IsNull())
+                return null;
+
+            //Tranform the point to the local coordinates of the ellipse
+            //After tranformation can see it as the ellipse centred in the origin with first axis long global x and second axis along global y
+            Cartesian coordinateSystem = Create.CartesianCoordinateSystem(ellipse.Centre, ellipse.Axis1, ellipse.Axis2);
+            TransformMatrix transform = Create.OrientationMatrixLocalToGlobal(coordinateSystem);
```

Comment:
Sanity check: shouldn't it be global to local, taken we are transforming from global to local?

Reply:
Ha, yeah, these always confuse me as well, but, no, should be like this. The transformation of LocalToGlobal would put the ellipse in the world origin, with Axis1 aligned with global X and Axis2 aligned with global Y. We do not need to make that transformation for the ellipse itself, as we can just assume it is there, but applying the same transformation to the point will essentially put it in the same reference R3, where it can be related to the world system as if it was the ellipse system.

Also, if it should be the other way around, the test scripts would fail miserably! :D (And in all honestly, might have had it the wrong way around first, with it failing, and then switching 😄 )

Reply:
This is... confusing 😃 but it only proves intuition is not always the best advisor - I took it wrong based on the name itself.